In [1]:
import pandas as pd
import numpy as np

In [2]:
ruta = "/Users/sandramartinez/Desktop/Proyectos_Data/TFM-EasyMoney/grupo-1-dscesp-0226/data/"

df_sal = pd.read_csv(ruta + "sales.csv", index_col=0)
df_cs  = pd.read_csv(ruta + "customer_sociodemographics.csv", index_col=0)
df_cca = pd.read_csv(ruta + "customer_commercial_activity.csv", index_col=0)
df_cp  = pd.read_csv(ruta + "customer_products.csv", index_col=0)
df_prd = pd.read_csv(ruta + "product_description.csv", index_col=0)

In [3]:
# Un cliente con al menos 3 compras
cid_ejemplo = df_sal["cid"].value_counts()[lambda s: s >= 3].index[0]

df_sal[df_sal["cid"] == cid_ejemplo].sort_values("month_sale")

,pk_sale,cid,month_sale,product_ID,net_margin
28487,35153,1387215,2018-03-01,2234,69.1
47537,54203,1387215,2018-03-01,1364,3746.6
140557,147223,1387215,2018-03-01,4657,62.7
1690,8356,1387215,2018-04-01,2335,1700.7
15154,21820,1387215,2018-04-01,2312,65.0
64988,71654,1387215,2018-04-01,2235,65.7
78450,85116,1387215,2018-04-01,8871,44.5
107765,114431,1387215,2018-04-01,3819,68.7
6085,12751,1387215,2018-07-01,2673,1260.1
5159,11825,1387215,2019-02-01,1119,1001.1


In [4]:
df_cp[df_cp["pk_cid"] == cid_ejemplo].sort_values("pk_partition")

,pk_cid,pk_partition,short_term_deposit,loans,mortgage,funds,securities,long_term_deposit,em_account_pp,credit_card,payroll,pension_plan,payroll_account,emc_account,debit_card,em_account_p,em_acount
1667901,1387215,2018-03,0,0,0,0,0,0,0,0,1.0,1.0,0,0,0,0,1
2305122,1387215,2018-04,1,0,0,0,0,0,0,1,1.0,1.0,1,1,1,0,0
2727756,1387215,2018-05,1,0,0,0,0,0,0,1,1.0,1.0,1,1,1,0,0
3357237,1387215,2018-06,1,0,0,0,0,0,0,0,1.0,1.0,1,1,1,0,0
4561710,1387215,2018-07,0,0,0,0,1,0,0,1,1.0,1.0,1,1,1,0,0
5077214,1387215,2018-08,0,0,0,0,0,0,0,1,1.0,1.0,1,1,1,0,0
6224434,1387215,2018-09,0,0,0,0,0,0,0,0,1.0,1.0,1,1,1,0,0
6590990,1387215,2018-10,0,0,0,0,0,0,0,0,1.0,1.0,1,1,1,0,0
7820090,1387215,2018-11,0,0,0,0,0,0,0,0,1.0,1.0,1,1,1,0,0
9013628,1387215,2018-12,0,0,0,0,0,0,0,1,1.0,1.0,1,1,1,0,0


Para el cliente 1387215:

n_compras = 10 (una por fila) 


meses_con_compra = 4 (2018‑03, 2018‑04, 2018‑07, 2019‑02) 

margen_total = 8.084,3 € (suma de net_margin) 

In [5]:
df_cca[df_cca["pk_cid"] == cid_ejemplo].sort_values("pk_partition")[["pk_partition", "active_customer", "segment"]]

,pk_partition,active_customer,segment
1667901,2018-03,1.0,01 - TOP
2305122,2018-04,1.0,01 - TOP
2727756,2018-05,1.0,01 - TOP
3357237,2018-06,1.0,01 - TOP
4561710,2018-07,1.0,01 - TOP
5077214,2018-08,1.0,01 - TOP
6224434,2018-09,1.0,01 - TOP
6590990,2018-10,1.0,01 - TOP
7820090,2018-11,1.0,01 - TOP
9013628,2018-12,1.0,01 - TOP


657826

In [7]:
sin_compras = df_clientes[df_clientes["n_compras"] == 0]

print("Sin compras y en la base desde el inicio (17 meses):",
      (sin_compras["antiguedad_meses"] == 17).sum())
print("Sin compras y entraron durante el periodo:",
      (sin_compras["antiguedad_meses"] < 17).sum())

# ¿Cuántos productos tienen los que entraron durante el periodo sin compras?
cols_prod = [c for c in df_cp.columns if c not in ("pk_cid", "pk_partition", "mes")]
nuevos_sin_compra = sin_compras[sin_compras["antiguedad_meses"] < 17]
print(nuevos_sin_compra[cols_prod].sum(axis=1).value_counts().sort_index())

NameError: name 'df_clientes' is not defined